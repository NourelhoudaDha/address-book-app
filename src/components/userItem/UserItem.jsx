import PropTypes from 'prop-types'

const UserItem = ({ thumbnail, firstName, lastName, username, email }) => (
  <li>
    <a className='hover:bg-light-blue-500 hover:border-transparent hover:shadow-lg group block rounded-lg p-4 border border-gray-200'>
      <img
        src={thumbnail}
        x-for='user in item.users'
        width='48'
        height='48'
        className='w-7 h-7 rounded-full bg-gray-100 border-2 border-white'
      />
      <dl className='grid sm:block lg:grid xl:block grid-cols-2 grid-rows-2 items-center'>
        <div>
          <dt className='sr-only'>Name</dt>
          <dd className='group-hover:text-white leading-6 font-medium text-black'>
            {firstName + ' ' + lastName}
          </dd>
        </div>
        <div>
          <dt className='sr-only'>Username</dt>
          <dd className='group-hover:text-light-blue-200 text-sm font-medium sm:mb-4 lg:mb-0 xl:mb-4'>
            {username}
          </dd>
        </div>
        <div>
          <dt className='sr-only'>Email</dt>
          <dd className='group-hover:text-light-blue-200 text-sm font-medium sm:mb-4 lg:mb-0 xl:mb-4'>
            {email}
          </dd>
        </div>
      </dl>
    </a>
  </li>
)

UserItem.propTypes = {
  thumbnail: PropTypes.string,
  firstName: PropTypes.string,
  lastName: PropTypes.string,
  username: PropTypes.string,
  email: PropTypes.string,
}
export default UserItem
