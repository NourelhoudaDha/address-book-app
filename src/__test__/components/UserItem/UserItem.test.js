import { render, cleanup } from '@testing-library/react'
import UserItem from '../../../components/userItem/UserItem.jsx'
const mockThumbnail = 'https://randomuser.me/api/portraits/thumb/women/43.jpg'
const mockFirstName = 'Nour Elhouda'
const mockLastName = 'Dhaouadi'
const mockUsername = 'username'
const mockEmail = 'dhaouadi.nour.elhouda@gmail.com'
afterEach(cleanup)
describe('UserItem', () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(
      <UserItem
        thumbnail={mockThumbnail}
        firstName={mockFirstName}
        lastName={mockLastName}
        username={mockUsername}
        email={mockEmail}
      />
    )
    expect(
      asFragment(
        <UserItem
          thumbnail={mockThumbnail}
          firstName={mockFirstName}
          lastName={mockLastName}
          username={mockUsername}
          email={mockEmail}
        />
      )
    ).toMatchSnapshot()
  })
})
