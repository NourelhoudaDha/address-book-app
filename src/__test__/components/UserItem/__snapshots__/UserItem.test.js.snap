// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserItem should take a snapshot 1`] = `
<DocumentFragment>
  <li>
    <a
      class="hover:bg-light-blue-500 hover:border-transparent hover:shadow-lg group block rounded-lg p-4 border border-gray-200"
    >
      <img
        class="w-7 h-7 rounded-full bg-gray-100 border-2 border-white"
        height="48"
        src="https://randomuser.me/api/portraits/thumb/women/43.jpg"
        width="48"
        x-for="user in item.users"
      />
      <dl
        class="grid sm:block lg:grid xl:block grid-cols-2 grid-rows-2 items-center"
      >
        <div>
          <dt
            class="sr-only"
          >
            Name
          </dt>
          <dd
            class="group-hover:text-white leading-6 font-medium text-black"
          >
            Nour Elhouda Dhaouadi
          </dd>
        </div>
        <div>
          <dt
            class="sr-only"
          >
            Username
          </dt>
          <dd
            class="group-hover:text-light-blue-200 text-sm font-medium sm:mb-4 lg:mb-0 xl:mb-4"
          >
            username
          </dd>
        </div>
        <div>
          <dt
            class="sr-only"
          >
            Email
          </dt>
          <dd
            class="group-hover:text-light-blue-200 text-sm font-medium sm:mb-4 lg:mb-0 xl:mb-4"
          >
            dhaouadi.nour.elhouda@gmail.com
          </dd>
        </div>
      </dl>
    </a>
  </li>
</DocumentFragment>
`;
